### PR TITLE
Use new API endpoint for scheduled job stats

### PR DESCRIPTION
**Pull request by quis**
Depends on:
- [x] https://github.com/alphagov/notifications-api/pull/2984

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1815,6 +1815,15 @@ def _get_jobs(service_id, limit_days=None, statuses=None, page=1):
     return mocker.patch('app.job_api_client.get_jobs', side_effect=_get_jobs)
 
 
+@pytest.fixture(scope='function')
+def mock_get_scheduled_job_stats(mocker, api_user_active):
+    return mocker.patch('app.job_api_client.get_scheduled_job_stats', return_value={
+        # These values match the return value of `mock_get_jobs`
```
Comment (idavidmcdonald) [inline tests/conftest.py:1821]: Helpful comment

**Review by idavidmcdonald — APPROVED** [review]
Looks good. Some optional comments for you

File: app/models/service.py [hunk 2]
```diff
@@ -151,6 +151,12 @@ def scheduled_jobs(self):
             return []
         return ScheduledJobs(self.id)
 
+    @cached_property
+    def scheduled_job_stats(self):
+        if not self.has_jobs:
```
[the remark this answers is not in the document]
Comment (quis) [reply app/models/service.py:156]:
```suggestion
        # This checks a cached value in Redis so is an optimisation for services with no jobs
        if not self.has_jobs:
```

File: tests/app/main/views/test_dashboard.py [hunk 3]
```diff
@@ -852,6 +856,52 @@ def test_should_show_upcoming_jobs_on_dashboard(
...
+    mock_get_usage,
+    mock_get_free_sms_fragment_limit,
+    mock_get_inbound_sms_summary,
+    mock_get_returned_letter_statistics_with_no_returned_letters,
+):
+    page = client_request.get(
+        'main.service_dashboard',
+        service_id=SERVICE_ONE_ID,
+    )
+    mock_has_no_jobs.assert_called_once_with(SERVICE_ONE_ID)
```
[the remark this answers is not in the document]
Comment (quis) [reply tests/app/main/views/test_dashboard.py:901]: Added in eb58851e9

**Pull request merged**